### ISSUE TITLE
Reproduction of RefGPT on other model

### ISSUE THREAD

**Issue by FUTUREEEEEE** [mutonix/RefGPT#4]
Hi, thanks for sharing you great work. I wonder if following configuration is the same as you produce the RefGPT-Fact dataset (GPT-4)? as I want to follow your work to produce similar data but with other models.

```
python parallel_generate-refgpt-fact.py \
      --reference_filepaths wiki_demo.jsonl \
      --api_config api_config.jsonl \
      --save_filepath output.jsonl \
      --num_chat_to_generate 20000 \
      --language en \
      --assistant_word_count 300 \
      --human_word_count 100 \
      --num_turn_ratios 0 1 2 2 0 \
      --max_attempts 5 \
      --max_requests_per_minute 1000 \
      --max_tokens_per_minute 40000 
```

**Comment by mutonix** [comment]
yes but it is noted we use azure openai api, so its api format is different from the chatcompletion api. ---- Replied Message ---- From Xia Qiang ***@***.***> Date 11/20/2024 12:41 To ***@***.***> Cc ***@***.***> Subject [mutonix/RefGPT] Reproduction of RefGPT on other model (Issue #4) Hi, thanks for sharing you great work. I wonder if following configuration is the same as you produce the RefGPT-Fact dataset (GPT-4), as I want to follow your work to produce similar data but with other models. python parallel_generate-refgpt-fact.py \
      --reference_filepaths wiki_demo.jsonl \
      --api_config api_config.jsonl \
      --save_filepath output.jsonl \
      --num_chat_to_generate 20000 \
      --language en \
      --assistant_word_count 300 \
      --human_word_count 100 \
      --num_turn_ratios 0 1 2 2 0 \
      --max_attempts 5 \
      --max_requests_per_minute 1000 \
      --max_tokens_per_minute 40000

**Comment by tuqingwen** [comment]
非常感谢您的工作，请问assistant_word_count 和num_turn_ratios 代表什么意思？

**Comment by mutonix** [comment]
1. assistant_word_count
假如生成3轮对话，assistant_word_count = [200, 300, 400]，第1，2，3轮assistant回答字数分别为200，300，400字
2. num_turn_ratios
假如我想生存2，3，4轮的对话，比例分别为0.2，0.3，0.5，num_turn_ratios = [0.2, 0.3, 0.5]

**Comment by tuqingwen** [comment]
{"api-key": "sk-", "request_url": "https://api.deepseek.com/chat/completions"}
{"api-key": "sk-", "request_url": "https://api.deepseek.com/chat/completions"}
上述是api_config文件，运行脚本后出现以下问题
WARNING:root:Request 0 failed with Exception 0, message='Attempt to decode JSON with unexpected mimetype: ', url=URL('https://api.deepseek.com/chat/completions')

**Comment by mutonix** [comment]
[header](https://github.com/mutonix/RefGPT/blob/27cd00bb3b531dcd65859ff4005010476edfd53c/parallel_generate-refgpt-fact.py#L58)的地方可能需要改一下，可能deepseek api的header和OpenAI的不太一样

**Comment by tuqingwen** [comment]
> [header](https://github.com/mutonix/RefGPT/blob/27cd00bb3b531dcd65859ff4005010476edfd53c/parallel_generate-refgpt-fact.py#L58)的地方可能需要改一下，可能deepseek api的header和OpenAI的不太一样

谢谢提醒，我去改改
